common4j/src/main/com/microsoft/identity/common/java/interfaces/ICommonComponents.java

```diff
@@ -41,4 +42,44 @@
      * Get an encryption manager for storage layer.
      */
     IKeyAccessor getStorageEncryptionManager(@Nullable final ITelemetryCallback telemetryCallback);
+
+    /**
+     * Retrieve a name-value store with a given identifier.
+     *
+     * @param storeName The name of a new KeyValue store.
+     * @param clazz     The class of values in the name value store.
+     * @return a INameValueStorage instance based around data stored with the same storeName.
+     */
+    <T> INameValueStorage<T> getNameValueStore(String storeName, Class<T> clazz);
+
+    /**
+     * Retrieve a name-value store with a given identifier.
+     *
+     * @param storeName The name of a new KeyValue store. May not be null.
+     * @param helper    The key manager for the encryption.  May be null.
+     * @param clazz     The class of values in the name value store. May not be null.
+     * @return a INameValueStorage instance based around data stored with the same storeName.
+     */
+    <T> INameValueStorage<T> getEncryptedNameValueStore(String storeName, IKeyAccessor helper, Class<T> clazz);
+
+    /**
+     * Get a generic encrypted ISharedPreferencesFileManager with a given identifier.
+     *
+     * @param storeName The name of a new KeyValue store. May not be null.
+     * @param helper    The key manager for the encryption.  May not be null.
+     */
+    ISharedPreferencesFileManager getEncryptedFileStore(String storeName, IKeyAccessor helper);
+
+    /**
+     * Get a generic encrypted ISharedPreferencesFileManager with a given identifier.
+     *
+     * @param storeName The name of a new KeyValue store. May not be null.
+     */
+    ISharedPreferencesFileManager getFileStore(String storeName);
+
+    /**
+     * Unwrap the platform context being shadowed by this interface.
+     * @return the underlying implementation of the platform context that this is hiding.
+     */
+    T getPlatformContext();
```

Comment:
Hmm...I think this is good, just not sure what I'm going to return here for non-Android case

Reply:
This is mainly a crutch.

Reply:
For non-android, it's probably fine to return null here, until and unless we come up with an underlying platform context object for linux.

Reply:
What would be `T` in that case? `null` isn't a `T` I guess....would we need `Void`?

Reply:
Yes.

Reply:
Dropped this method.